Libs/vtkAddon/vtkAddonMathUtilities.cxx

```diff
@@ -116,3 +119,80 @@ void vtkAddonMathUtilities::GetOrientationMatrix(vtkMatrix4x4* source,
       }
     }
 }
+
+//----------------------------------------------------------------------------
+std::string vtkAddonMathUtilities::ToString(const vtkMatrix4x4* mat, const std::string delimiter, const std::string rowDelimiter)
+{
+  if (!mat)
+    {
+    return "";
+    }
+
+  std::stringstream ss;
+  for (int i = 0; i < 4; i++)
+    {
+    for (int j = 0; j < 4; j++)
+      {
+      ss << mat->GetElement(i, j);
+      ss << delimiter;
+      }
+    ss << rowDelimiter;
+    }
+
+  return ss.str();
+}
+
+//----------------------------------------------------------------------------
+bool vtkAddonMathUtilities::FromString(vtkMatrix4x4* mat, const std::string& str, const std::string delimiterExp)
+{
+  if (!mat)
+  {
+    return false;
+  }
+
+  // Convert expression into a stringstream to convert and parse into doubles for the matrix
+
+  // Parse the string using the regular expression
+  std::regex delimiterRegex(delimiterExp);
+  std::sregex_token_iterator itr(str.begin(), str.end(), delimiterRegex, -1); // Use matches as delimiters to split the string
+  std::sregex_token_iterator emptyItr;
+
+  // Put into stringstream
+  std::stringstream ss;
+  int numElements = 0;
+  while(itr!=emptyItr)
+    {
+    std::string valString( *itr );
+    itr++;
+    if (valString.empty())
+      {
+      continue; // Ignore empty strings - it just means we have back-to-back delimiters
+      }
+    ss << valString << " "; // Space is a delimiter that stringstream can handle
+    numElements++;
+    }
+
+  // Ensure the matrix is 1x1, 2x2, 3x3, or 4x4
+  double size = std::sqrt(numElements);
+  if (std::floor(size)!=size || size>4)
```

Comment:
Don't do floating point comparison without tolerance. It would be simpler to just check that numElements is 1, 4, 9, or 16.